### DISCUSSION TITLE
关于 MarkupExtension 的Converter 有没有方法让xaml中传递多个参数给一个属性，如方法中的params 那样？

### DISCUSSION THREAD

**Discussion by yoi102** [BYJRK/DotNet-Discussions#77] [Q&A]
我有一个需求：根据Enum的值实现一些控件显示和隐藏，但并非所有的值都单独控制显示或隐藏控件，有部分的值需要显示其中共同的控件，有没有比较好的方法实现EnumToVisibilityConverter?(使用DataTrigger来实现这功能需要写一大串代码。。。。。)；

![image](https://user-images.githubusercontent.com/77535233/230771128-2d45da22-2696-4df7-a20a-fa39e0e6979c.png)

**Comment by BYJRK** [comment]
你要不要考虑用template selector？

**Reply by yoi102** [reply]
https://learn.microsoft.com/zh-cn/dotnet/api/system.windows.controls.contentcontrol.contenttemplateselector?f1url=%3FappId%3DDev16IDEF1%26l%3DZH-CN%26k%3Dk(System.Windows.Controls.ContentControl.ContentTemplateSelector)%3Bk(VS.XamlEditor)%26rd%3Dtrue&view=windowsdesktop-7.0
我没怎么用过，说的是这个吗？
这个的话，由于只会返回一个 Template ，如果要显示共同的控件，将会要在多个 Tamplate 出现重复的控件。
如果为每个界面定制一个 TemplateSelector ，感觉也会不太值得。。。。

**Reply by yoi102** [reply]
不好意思理解错了，不太需要为定制 TemplateSelector 。。。。但共同要显示的控件，会重复出现在多个Template中。。。
